vsphere_plugin_common/__init__.py

```diff
@@ -322,9 +336,57 @@ def create_server(self,
                     vim.vm.device.VirtualDeviceSpec.Operation.remove
                 devices.append(nicspec)
 
+        port_groups, distributed_port_groups = self._get_port_group_names()
+
         for network in networks:
             network_name = network['name']
+            network_name_lower = network_name.lower()
             switch_distributed = network['switch_distributed']
+
+            # Check network exists and provide helpful message if it doesn't
+            # Note that we special-case alerting if switch_distributed appears
+            # to be set incorrectly.
+            # Use lowercase name for comparison as vSphere appears to be case
+            # insensitive for this.
+            if switch_distributed:
```

Comment:
It's been a fairly frequent cause of e-mailed issues in my recollection (which may be coloured by perception bias), so I wanted to make it all as clear as I could.

Even if end-users don't read it, at least it should make it take next to no cognitive effort for us to troubleshoot when those errors come up.